File: src/main/resources/static/js/service/collectionService.js
//服务层
app.service('collectionService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../collection/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../collection/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../collection/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../collection/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../collection/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../collection/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../collection/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: src/main/java/cdtu/wheretobuy/service/CollectionService.java
package cdtu.wheretobuy.service;
import java.util.List;
import cdtu.wheretobuy.pojo.Collection;

import cdtu.wheretobuy.pojo.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CollectionService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Collection> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Collection collection);
	
	
	/**
	 * 修改
	 */
	public void update(Collection collection);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Collection findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Collection collection, int pageNum, int pageSize);
	
}
